src/mgmt/rpc/jsonrpc/JsonRPCManager.cc

```diff
@@ -112,7 +111,7 @@ JsonRPCManager::Dispatcher::dispatch(Context const &ctx, specs::RPCRequestInfo c
   }
 
   // We have got a valid handler, we will now check if the context holds any restriction for this handler to be called.
-  if (auto errata = check_for_blockers(ctx, handler.get_options()); !errata.is_ok()) {
+  if (auto &&[errata, ec] = check_for_blockers(ctx, handler.get_options()); !errata.is_ok()) {
```

Comment:
The error was detected by this check, but the error code was never set. This fixes that as `ec`  now holds the error.